### PR TITLE
adding-utils-and-json-api-packages

### PR DESCRIPTION
**Pull request by Sorizen**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,7 +8,6 @@ packages/**/dist-browser/
 /doc/
 npm-debug.log*
 yarn-debug.log*
-yarn-error.log*
```
Comment (napalmpapalam) [inline .gitignore]: why did u remove it?

File: jest.config.base.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  testMatch: [
+    "**/__tests__/**/*.+(ts|tsx|js)",
+    "**/?(*.)+(spec|test).+(ts|tsx|js)"
+  ],
+  transform: {
+    "^.+\\.(ts|tsx)$": "ts-jest"
+  },
+  moduleNameMapper: {
+    "^@/(.*)": "<rootDir>/src/$1",
+    "^lodash-es$": "lodash"
```
Comment (napalmpapalam) [inline jest.config.base.js]: this possibly can be removed, you shouldn't use `lodash-es`, it wouldn't compile into CommonJS, so try to remove it or rename it to `lodash` if you resolve some errors

File: packages/utils/README.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
-# @dl/utils
-Typescript-based utils for Distributed Lab projects
+# @dl/json-api-client
```
Comment (napalmpapalam) [inline packages/utils/README.md]: it's not json-api-client

File: packages/utils/src/browser.ts [hunk 4]
```diff
@@ -5,6 +5,6 @@
 import * as utils from './index'
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-;(window as any).webKitUtils = utils
+  ;(window as any).webKitUtils = utils
```
Comment (napalmpapalam) [inline packages/utils/src/browser.ts]: ?

File: packages/utils/src/date.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+  OptionType,
+  OpUnitType,
+  UnitType,
+  ManipulateType,
+} from 'dayjs'
+import isSameOrAfter from 'dayjs/plugin/isSameOrAfter'
+import isSameOrBefore from 'dayjs/plugin/isSameOrBefore'
+
+type IsoDate = string
+type UnixDate = number
```
Comment (napalmpapalam) [inline packages/utils/src/date.ts]: move to `types/date.ts`

File: packages/utils/src/math.ts [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    const two = new BigNumber(b)
+
+    const result = one.plus(two)
+    return result.toFixed(DECIMALS)
+  }
+
+  static subtract(
+    a: string,
+    b: string,
+    ROUND_TYPE = ROUNDING_MODES.ROUND_UP,
```
Comment (napalmpapalam) [inline packages/utils/src/math.ts]: try to omit casting type with `as` in `BigNumber.config()`:

```ts
type RoundingMode = BigNumber.RoundingMode

static subtract(
...
ROUND_TYPE: RoundingMode  = ROUNDING_MODES.ROUND_UP
...
)
```

File: packages/utils/src/math.ts [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+      new BigNumber(a)
+      new BigNumber(b)
+      new BigNumber(c)
+    } catch (err) {
+      return false
+    }
+    return true
+  }
+
+  static get roundingModes(): typeof ROUNDING_MODES {
```
Comment (napalmpapalam) [inline packages/utils/src/math.ts]: use created type `RoundingModes`, as I wrote upwards

File: packages/utils/tsconfig.json [hunk 8]
```diff
@@ -7,7 +7,8 @@
       "@/*": ["src/*"]
     },
     "types": [
-      "node"
+      "node",
+      "jest"
     ],
     "lib": [
       "ES6",
```
Comment (napalmpapalam) [inline packages/utils/tsconfig.json:14]: Change log

**Review by napalmpapalam — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (Sorizen) [reply .gitignore]: returned

File: jest.config.base.js (hunk 2 above)
Comment (Sorizen) [reply jest.config.base.js]: removed

File: packages/utils/README.md (hunk 3 above)
Comment (Sorizen) [reply packages/utils/README.md]: returned

File: packages/utils/src/browser.ts (hunk 4 above)
Comment (Sorizen) [reply packages/utils/src/browser.ts]: returned

File: packages/utils/src/math.ts (hunk 6 above)
Comment (Sorizen) [reply packages/utils/src/math.ts]: done

File: packages/utils/src/math.ts (hunk 7 above)
Comment (Sorizen) [reply packages/utils/src/math.ts]: done

File: packages/utils/src/date.ts (hunk 5 above)
Comment (Sorizen) [reply packages/utils/src/date.ts]: done

File: packages/utils/types/date.ts [hunk 9]
```diff
@@ -0,0 +1,2 @@
+export type IsoDate = string // RFC3999Nano ISO Date String
```
Comment (napalmpapalam) [inline packages/utils/types/date.ts:1]: u should export it from `index.ts` from the current directory and then export it from `src/index.ts`, thus client-code be able to import type from the lib

File: packages/utils/CHANGELOG.md [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+## [Unreleased]
+
+### Added
+
+### Changed
+
+### Removed
+
+### Fixed
```
Comment (napalmpapalam) [inline packages/utils/CHANGELOG.md]: u don't have to add all `###*` if u don't have changes for it, just to 

```md
##[Unreleased]
### Under the hood changes
- Initiated project with basic utilities
```

**Review by napalmpapalam — CHANGES_REQUESTED** [review]

File: packages/utils/types/date.ts (hunk 9 above)
Comment (Sorizen) [reply packages/utils/types/date.ts:1]: done

File: packages/utils/CHANGELOG.md (hunk 10 above)
Comment (Sorizen) [reply packages/utils/CHANGELOG.md]: done

File: jest.config.base.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+  testMatch: [
+    "**/__tests__/**/*.+(ts|tsx|js)",
+    "**/?(*.)+(spec|test).+(ts|tsx|js)"
+  ],
+  transform: {
+    "^.+\\.(ts|tsx)$": "ts-jest"
+  },
+  moduleNameMapper: {
+    "^@/(.*)": "<rootDir>/src/$1",
+  },
```
Comment (napalmpapalam) [inline jest.config.base.js:15]: are u sure that this cannot be populated from base config or at least any of these fields?

File: packages/json-api-client/CHANGELOG.md [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres
+to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+Please check our [developers guide](https://gitlab.com/tokend/developers-guide)
+for further information about branching and tagging conventions.
+
+##[Unreleased]
```
Comment (napalmpapalam) [inline packages/json-api-client/CHANGELOG.md]: let's initiate packages, for now, with `1.0.0` version, also add do the same for `@dl/utils`

```markdown
## [1.0.0] - 2022-09-13
### Under the hood changes
- Initiated project with basic functionality
...
```

File: packages/json-api-client/package.json [hunk 13]
```diff
@@ -20,10 +20,12 @@
   "typedoc": {
     "entryPoint": "./src/index.ts",
     "readmeFile": "./README.md",
-    "displayName": "@dl/json-api-client"
+    "displayName": "@dl/utils"
```
Comment (napalmpapalam) [inline packages/json-api-client/package.json]: wtf?

File: packages/json-api-client/package.json [hunk 14]
```diff
@@ -43,8 +45,15 @@
     "@types/node": "^18.0.6",
     "babel-plugin-root-import": "^6.6.0",
     "esbuild": "^0.14.49",
+    "jest": "^29.0.1",
     "node-stdlib-browser": "^1.2.0",
     "tsc-alias": "^1.7.0",
     "typescript": "^4.7.4"
+  },
+  "dependencies": {
+    "@types/lodash": "^4.14.184",
```
Comment (napalmpapalam) [inline packages/json-api-client/package.json]: this should be in `devDependencies` section

File: packages/json-api-client/src/const/client.const.ts [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+  [CLIENT_API_ENDPOINT_TYPES.core]: 'https://api.dev.rarify.tech/core',
+  [CLIENT_API_ENDPOINT_TYPES.public]: 'https://api.dev.rarify.tech/public',
+  [CLIENT_API_ENDPOINT_TYPES.data]: 'https://api.dev.rarify.tech/data',
+} as const
+
+export const CLIENT_PROD_API_CLUSTER_BASE_URLS = {
+  [CLIENT_API_ENDPOINT_TYPES.core]: 'https://api.rarify.tech/core',
+  [CLIENT_API_ENDPOINT_TYPES.public]: 'https://api.rarify.tech/public',
+  [CLIENT_API_ENDPOINT_TYPES.data]: 'https://api.rarify.tech/data',
+} as const
```
Comment (napalmpapalam) [inline packages/json-api-client/src/const/client.const.ts]: 1. u don't need this
2. I said don't push unrelated stuff to Github

File: packages/json-api-client/src/enums/application.ts [hunk 16]
```diff
@@ -0,0 +1,11 @@
...
+  development = 'dev',
+  staging = 'staging',
+  production = 'production',
+}
+
+export enum APPLICATION_STATE_VALUES {
+  active = 1,
+  paused = 2,
+  deleted = 3,
+}
```
Comment (napalmpapalam) [inline packages/json-api-client/src/enums/application.ts]: how it's related with json api http client

File: packages/json-api-client/src/enums/client.ts [hunk 17]
```diff
@@ -0,0 +1,5 @@
+export enum CLIENT_API_ENDPOINT_TYPES {
+  core = 'core',
+  public = 'public',
+  data = 'data',
+}
```
Comment (napalmpapalam) [inline packages/json-api-client/src/enums/client.ts]: how it's related with json api http client

File: packages/json-api-client/src/enums/data.ts [hunk 18]
```diff
@@ -0,0 +1,8 @@
+export enum COLLECTION_PRIMARY_INTERFACE {
+  unknown = 'unknown',
+  erc721 = 'erc_721',
+  erc1155 = 'erc_1155',
+  erc20 = 'erc_20',
+  needRecheck = 'need_recheck',
+  nativeToken = 'native_token',
+}
```
Comment (napalmpapalam) [inline packages/json-api-client/src/enums/data.ts]: how it's related with json api http client

File: packages/json-api-client/src/enums/http.ts [hunk 19]
```diff
@@ -0,0 +1,16 @@
+export enum HTTP_STATUS_CODES {
+  badRequest = 400,
+  unauthorized = 401,
+  forbidden = 403,
+  notFound = 404,
+  conflict = 409,
+  internalError = 500,
+}
```
Comment (napalmpapalam) [inline packages/json-api-client/src/enums/http.ts]: https://www.npmjs.com/package/http-status

File: packages/json-api-client/src/enums/http.ts [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+
+export enum HTTP_METHODS {
+  GET = 'GET',
+  POST = 'POST',
+  PATCH = 'PATCH',
+  PUT = 'PUT',
+  DELETE = 'DELETE',
+}
```
Comment (napalmpapalam) [inline packages/json-api-client/src/enums/http.ts:7]: https://www.npmjs.com/package/http-methods-constants

File: packages/json-api-client/src/enums/identity.ts [hunk 21]
```diff
@@ -0,0 +1,4 @@
+export enum IDENTITY_STATES {
```
Comment (napalmpapalam) [inline packages/json-api-client/src/enums/identity.ts]: how it's related with json api http client

File: packages/json-api-client/src/enums/network.ts [hunk 22]
```diff
@@ -0,0 +1,9 @@
+export enum NETWORK_PROTOCOLS {
+  ethereum = 'Ethereum',
+}
+
+export enum NETWORKS {
+  polygon = 'polygon',
+  ethereum = 'ethereum',
+  rinkeby = 'rinkeby',
+}
```
Comment (napalmpapalam) [inline packages/json-api-client/src/enums/network.ts]: how it's related with json api http client

File: packages/json-api-client/src/enums/transaction.ts [hunk 23]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * "failed" - transaction failed to complete. Refer to failure_reason for
+ * details (final state).
+ */
+export enum TRANSACTION_STATES {
+  pending = 'pending',
+  applied = 'applied',
+  built = 'built',
+  submitted = 'submitted',
+  failed = 'failed',
```
Comment (napalmpapalam) [inline packages/json-api-client/src/enums/transaction.ts]: how it's related with json api http client

File: packages/json-api-client/src/helpers/type-checks.ts [hunk 24]
```diff
@@ -0,0 +1,35 @@
...
+
+export const isIdentity = (owner: any): owner is Identity => {
+  return Boolean(owner.basics)
+}
+
+export const isNetwork = (network: any): network is Network => {
+  return Boolean(network.protocol)
+}
+
+export const isMetaSellOrderArray = (
```
Comment (napalmpapalam) [inline packages/json-api-client/src/helpers/type-checks.ts]: how it's related with json api http client

File: packages/json-api-client/src/response.ts [hunk 25]
```diff
@@ -0,0 +1,157 @@
...
+   */
+  get isLinksExist(): boolean {
+    return Boolean(this._links) && !isEmpty(this._links)
+  }
+
+  /**
+   * Parses and unwraps response data.
+   */
+  private _parseResponse(raw: AxiosResponse, needRaw: boolean) {
+    if (raw.status === 204 || raw.status === 205) {
```
Comment (napalmpapalam) [inline packages/json-api-client/src/response.ts]: use constants

File: packages/json-api-client/src/response.ts [hunk 26]
```diff
@@ -0,0 +1,157 @@
...
+
+    let intersection = ''
+
+    for (const char of link) {
+      if (baseUrl.endsWith(intersection + char)) {
+        intersection += char
+        break
+      } else {
+        intersection += char
+      }
```
Comment (napalmpapalam) [inline packages/json-api-client/src/response.ts]:
```js
     intersection += char
     if (baseUrl.endsWith(intersection + char)) break
```

File: packages/json-api-client/src/types/callers.ts [hunk 27]
```diff
@@ -0,0 +1,219 @@
+import {
```
Comment (napalmpapalam) [inline packages/json-api-client/src/types/callers.ts]: this file should be deleted

File: packages/json-api-client/src/types/client.ts [hunk 28]
```diff
@@ -0,0 +1,14 @@
+import {
```
Comment (napalmpapalam) [inline packages/json-api-client/src/types/client.ts]: this file should be deleted

**Review by napalmpapalam — CHANGES_REQUESTED** [review]

File: packages/json-api-client/src/browser.ts [hunk 29]
```diff
@@ -2,9 +2,9 @@
...
  */
-import * as jsonApiClient from './index'
+import * as JsonApi from './index'
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-;(window as any).webKitJsonApiClient = jsonApiClient
+;(window as any).webKitJsonApi = JsonApi
 
-console.warn('Web Kit JSON API Client was added to the window object.')
+console.warn('Web Kit Utils was added to the window object.')
```
Comment (napalmpapalam) [inline packages/json-api-client/src/browser.ts]: ????

**Review by napalmpapalam — CHANGES_REQUESTED** [review]

File: packages/json-api-client/package.json (hunk 13 above)
Comment (Sorizen) [reply packages/json-api-client/package.json]: done

File: packages/json-api-client/package.json (hunk 14 above)
Comment (Sorizen) [reply packages/json-api-client/package.json]: done

File: packages/json-api-client/src/const/client.const.ts (hunk 15 above)
Comment (Sorizen) [reply packages/json-api-client/src/const/client.const.ts]: done

File: packages/json-api-client/src/enums/application.ts (hunk 16 above)
Comment (Sorizen) [reply packages/json-api-client/src/enums/application.ts]: done

File: packages/json-api-client/src/enums/client.ts (hunk 17 above)
Comment (Sorizen) [reply packages/json-api-client/src/enums/client.ts]: done

File: packages/json-api-client/src/enums/data.ts (hunk 18 above)
Comment (Sorizen) [reply packages/json-api-client/src/enums/data.ts]: done

File: packages/json-api-client/src/enums/identity.ts (hunk 21 above)
Comment (Sorizen) [reply packages/json-api-client/src/enums/identity.ts]: done

File: packages/json-api-client/src/enums/network.ts (hunk 22 above)
Comment (Sorizen) [reply packages/json-api-client/src/enums/network.ts]: done

File: packages/json-api-client/src/enums/transaction.ts (hunk 23 above)
Comment (Sorizen) [reply packages/json-api-client/src/enums/transaction.ts]: done

File: packages/json-api-client/src/helpers/type-checks.ts (hunk 24 above)
Comment (Sorizen) [reply packages/json-api-client/src/helpers/type-checks.ts]: done

File: packages/json-api-client/src/response.ts (hunk 26 above)
Comment (Sorizen) [reply packages/json-api-client/src/response.ts]: done

File: packages/json-api-client/src/types/callers.ts (hunk 27 above)
Comment (Sorizen) [reply packages/json-api-client/src/types/callers.ts]: done

File: packages/json-api-client/src/types/client.ts (hunk 28 above)
Comment (Sorizen) [reply packages/json-api-client/src/types/client.ts]: done

File: packages/json-api-client/src/browser.ts (hunk 29 above)
Comment (Sorizen) [reply packages/json-api-client/src/browser.ts]: done

File: packages/json-api-client/src/response.ts (hunk 25 above)
Comment (Sorizen) [reply packages/json-api-client/src/response.ts]: done

File: packages/json-api-client/src/enums/http.ts (hunk 19 above)
Comment (Sorizen) [reply packages/json-api-client/src/enums/http.ts]: done

File: packages/json-api-client/src/enums/http.ts (hunk 20 above)
Comment (Sorizen) [reply packages/json-api-client/src/enums/http.ts:7]: done

File: jest.config.base.js (hunk 11 above)
Comment (Sorizen) [reply jest.config.base.js:15]: done

File: packages/json-api-client/CHANGELOG.md (hunk 12 above)
Comment (Sorizen) [reply packages/json-api-client/CHANGELOG.md]: done

File: packages/json-api-client/package.json [hunk 30]
```diff
@@ -8,8 +8,8 @@
   },
   "main": "./lib/cjs/index.js",
   "module": "./lib/esm/index.js",
-  "browser": "./lib/json-api-client.min.js",
-  "unpkg": "./lib/json-api-client.min.js",
+  "browser": "./lib/utils.min.js",
```
Comment (napalmpapalam) [inline packages/json-api-client/package.json]: why a u doing this ?

File: packages/json-api-client/src/browser.ts (hunk 29 above)
Comment (napalmpapalam) [reply packages/json-api-client/src/browser.ts]: what is done? there are still "Utils" in the log

File: packages/json-api-client/src/json-api.ts [hunk 31]
```diff
@@ -0,0 +1,214 @@
...
+   *
+   *  For more details look Axios Request config:
+   *  {@link https://github.com/axios/axios#request-config}
+   */
+  get baseUrl(): Url {
+    return this._baseUrl
+  }
+
+  /**
+   * Rarify manages authentication and ownership within its system via bearer
```
Comment (napalmpapalam) [inline packages/json-api-client/src/json-api.ts]: remove rarify mention

File: packages/json-api-client/src/json-api.ts [hunk 32]
```diff
@@ -0,0 +1,214 @@
...
+   * tokens and internal identities.
+   *
+   * All endpoints in the Rarify API use bearer authToken authentication.
```
Comment (napalmpapalam) [inline packages/json-api-client/src/json-api.ts]: .

File: packages/json-api-client/src/test/mocks/json-api-response-parsed.json [hunk 33]
```diff
@@ -0,0 +1,335 @@
...
+          "id": "b66ce19f-f742-4e53-9d96-ff63d8c81bab",
+          "attributes": {},
+          "created_at": "2021-12-17T13:37:33.965001Z",
+          "description": "asdasd",
+          "image_url": "https://my-fake-url.com/example",
+          "name": "test",
+          "updated_at": "2021-12-17T13:37:33.965001Z",
+          "network": {
+            "type": "networks",
+            "id": "rarinet"
```
Comment (napalmpapalam) [inline packages/json-api-client/src/test/mocks/json-api-response-parsed.json]: remove rarify mention

File: packages/json-api-client/src/test/mocks/json-api-response-parsed.json [hunk 34]
```diff
@@ -0,0 +1,335 @@
...
+      "meta_sell_order"
+    ]
+  },
+  {
+    "type": "meta-buy-order",
+    "id": "03980a87-9afb-4952-86c6-674eaf555e51",
+    "amount": 1,
+    "claimable_at": "2022-01-25T14:29:17.230663Z",
+    "created_at": "2022-02-01T08:39:08.900574Z",
+    "email": "loktionov.kh@gmail.com",
```
Comment (napalmpapalam) [inline packages/json-api-client/src/test/mocks/json-api-response-parsed.json]: replace with foo@bar.com

File: packages/json-api-client/src/test/mocks/json-api-response-parsed.json [hunk 35]
```diff
@@ -0,0 +1,335 @@
...
+          "type": "contracts",
+          "id": "6b1658e4-d3cf-426d-a10d-8edad1055c3b"
+        },
+        "metadata": {
+          "type": "metadata",
+          "id": "c4b6da1f-6ca1-43ce-bea8-515cdf29cd59",
+          "attributes": {},
+          "created_at": "2021-12-29T14:36:01.796775Z",
+          "description": "asd",
+          "image_url": "https://rarify-user-content-dev.s3.amazonaws.com/2b3a17df-932d-4421-b03c-b05cebc7eae5",
```
Comment (napalmpapalam) [inline packages/json-api-client/src/test/mocks/json-api-response-parsed.json]: `https://my-fake-url.com/2b3a17df-932d-4421-b03c-b05cebc7eae`

File: packages/json-api-client/src/test/mocks/json-api-response-parsed.json [hunk 36]
```diff
@@ -0,0 +1,335 @@
...
+      "meta_sell_order"
+    ]
+  },
+  {
+    "type": "meta-buy-order",
+    "id": "497314c0-3d99-4a22-aebb-0c60913324da",
+    "amount": 1,
+    "claimable_at": "2022-01-25T14:29:17.230663Z",
+    "created_at": "2022-01-25T08:31:18.15616Z",
+    "email": "loktionov.semen.kh@gmail.com",
```
Comment (napalmpapalam) [inline packages/json-api-client/src/test/mocks/json-api-response-parsed.json]: /

File: packages/json-api-client/src/test/mocks/json-api-response-parsed.json [hunk 37]
```diff
@@ -0,0 +1,335 @@
...
+          "type": "contracts",
+          "id": "6b1658e4-d3cf-426d-a10d-8edad1055c3b"
+        },
+        "metadata": {
+          "type": "metadata",
+          "id": "c4b6da1f-6ca1-43ce-bea8-515cdf29cd59",
+          "attributes": {},
+          "created_at": "2021-12-29T14:36:01.796775Z",
+          "description": "asd",
+          "image_url": "https://rarify-user-content-dev.s3.amazonaws.com/2b3a17df-932d-4421-b03c-b05cebc7eae5",
```
Comment (napalmpapalam) [inline packages/json-api-client/src/test/mocks/json-api-response-parsed.json]: /

File: packages/json-api-client/src/test/mocks/json-api-response-parsed.json [hunk 38]
```diff
@@ -0,0 +1,335 @@
...
+      "meta_sell_order"
+    ]
+  },
+  {
+    "type": "meta-buy-order",
+    "id": "c5b5ca54-1760-43a5-a8ef-b48aad0e30d7",
+    "amount": 1,
+    "claimable_at": "2022-01-25T14:29:17.230663Z",
+    "created_at": "2022-01-25T08:29:07.054527Z",
+    "email": "loktionov.semen.kh@gmail.com",
```
Comment (napalmpapalam) [inline packages/json-api-client/src/test/mocks/json-api-response-parsed.json]: /

File: packages/json-api-client/src/test/mocks/json-api-response-parsed.json [hunk 39]
```diff
@@ -0,0 +1,335 @@
...
+          "type": "contracts",
+          "id": "6b1658e4-d3cf-426d-a10d-8edad1055c3b"
+        },
+        "metadata": {
+          "type": "metadata",
+          "id": "c4b6da1f-6ca1-43ce-bea8-515cdf29cd59",
+          "attributes": {},
+          "created_at": "2021-12-29T14:36:01.796775Z",
+          "description": "asd",
+          "image_url": "https://rarify-user-content-dev.s3.amazonaws.com/2b3a17df-932d-4421-b03c-b05cebc7eae5",
```
Comment (napalmpapalam) [inline packages/json-api-client/src/test/mocks/json-api-response-parsed.json]: /

File: packages/json-api-client/src/test/mocks/json-api-response-raw.json [hunk 40]
```diff
@@ -0,0 +1,259 @@
...
+      }
+    },
+    {
+      "id": "03980a87-9afb-4952-86c6-674eaf555e51",
+      "type": "meta-buy-order",
+      "attributes": {
+        "amount": 1,
+        "claimable_at": "2022-01-25T14:29:17.230663Z",
+        "created_at": "2022-02-01T08:39:08.900574Z",
+        "email": "loktionov.kh@gmail.com",
```
Comment (napalmpapalam) [inline packages/json-api-client/src/test/mocks/json-api-response-raw.json]: /

File: packages/json-api-client/src/test/mocks/json-api-response-raw.json [hunk 41]
```diff
@@ -0,0 +1,259 @@
...
+      }
+    },
+    {
+      "id": "497314c0-3d99-4a22-aebb-0c60913324da",
+      "type": "meta-buy-order",
+      "attributes": {
+        "amount": 1,
+        "claimable_at": "2022-01-25T14:29:17.230663Z",
+        "created_at": "2022-01-25T08:31:18.15616Z",
+        "email": "loktionov.semen.kh@gmail.com",
```
Comment (napalmpapalam) [inline packages/json-api-client/src/test/mocks/json-api-response-raw.json]: /

File: packages/json-api-client/src/test/mocks/json-api-response-raw.json [hunk 42]
```diff
@@ -0,0 +1,259 @@
...
+      }
+    },
+    {
+      "id": "c5b5ca54-1760-43a5-a8ef-b48aad0e30d7",
+      "type": "meta-buy-order",
+      "attributes": {
+        "amount": 1,
+        "claimable_at": "2022-01-25T14:29:17.230663Z",
+        "created_at": "2022-01-25T08:29:07.054527Z",
+        "email": "loktionov.semen.kh@gmail.com",
```
Comment (napalmpapalam) [inline packages/json-api-client/src/test/mocks/json-api-response-raw.json]: /

File: packages/json-api-client/src/test/mocks/json-api-response-raw.json [hunk 43]
```diff
@@ -0,0 +1,259 @@
...
+        "created_at": "2021-12-17T13:37:33.965001Z",
+        "description": "asdasd",
+        "image_url": "https://my-fake-url.com/example",
+        "name": "test",
+        "updated_at": "2021-12-17T13:37:33.965001Z"
+      },
+      "relationships": {
+        "network": {
+          "data": {
+            "id": "rarinet",
```
Comment (napalmpapalam) [inline packages/json-api-client/src/test/mocks/json-api-response-raw.json]: /

File: packages/json-api-client/src/test/mocks/json-api-response-raw.json [hunk 44]
```diff
@@ -0,0 +1,259 @@
...
+      }
+    },
+    {
+      "id": "c4b6da1f-6ca1-43ce-bea8-515cdf29cd59",
+      "type": "metadata",
+      "attributes": {
+        "attributes": {},
+        "created_at": "2021-12-29T14:36:01.796775Z",
+        "description": "asd",
+        "image_url": "https://rarify-user-content-dev.s3.amazonaws.com/2b3a17df-932d-4421-b03c-b05cebc7eae5",
```
Comment (napalmpapalam) [inline packages/json-api-client/src/test/mocks/json-api-response-raw.json]: /

File: packages/json-api-client/src/test/mocks/json-api-response-raw.json [hunk 45]
```diff
@@ -0,0 +1,259 @@
...
+        "name": "response",
+        "updated_at": "2021-12-29T14:36:01.796775Z"
+      },
+      "relationships": {
+        "network": {
+          "data": {
+            "id": "rarinet",
```
Comment (napalmpapalam) [inline packages/json-api-client/src/test/mocks/json-api-response-raw.json]: /

File: packages/json-api-client/src/test/mocks/json-api-response-without-links-raw.json [hunk 46]
```diff
@@ -0,0 +1,252 @@
...
+      }
+    },
+    {
+      "id": "03980a87-9afb-4952-86c6-674eaf555e51",
+      "type": "meta-buy-order",
+      "attributes": {
+        "amount": 1,
+        "claimable_at": "2022-01-25T14:29:17.230663Z",
+        "created_at": "2022-02-01T08:39:08.900574Z",
+        "email": "loktionov.kh@gmail.com",
```
Comment (napalmpapalam) [inline packages/json-api-client/src/test/mocks/json-api-response-without-links-raw.json]: /

File: packages/json-api-client/src/test/mocks/json-api-response-without-links-raw.json [hunk 47]
```diff
@@ -0,0 +1,252 @@
...
+      }
+    },
+    {
+      "id": "497314c0-3d99-4a22-aebb-0c60913324da",
+      "type": "meta-buy-order",
+      "attributes": {
+        "amount": 1,
+        "claimable_at": "2022-01-25T14:29:17.230663Z",
+        "created_at": "2022-01-25T08:31:18.15616Z",
+        "email": "loktionov.semen.kh@gmail.com",
```
Comment (napalmpapalam) [inline packages/json-api-client/src/test/mocks/json-api-response-without-links-raw.json]: /

File: packages/json-api-client/src/test/mocks/json-api-response-without-links-raw.json [hunk 48]
```diff
@@ -0,0 +1,252 @@
...
+      }
+    },
+    {
+      "id": "c5b5ca54-1760-43a5-a8ef-b48aad0e30d7",
+      "type": "meta-buy-order",
+      "attributes": {
+        "amount": 1,
+        "claimable_at": "2022-01-25T14:29:17.230663Z",
+        "created_at": "2022-01-25T08:29:07.054527Z",
+        "email": "loktionov.semen.kh@gmail.com",
```
Comment (napalmpapalam) [inline packages/json-api-client/src/test/mocks/json-api-response-without-links-raw.json]: /

File: packages/json-api-client/src/test/mocks/json-api-response-without-links-raw.json [hunk 49]
```diff
@@ -0,0 +1,252 @@
...
+      }
+    },
+    {
+      "id": "c4b6da1f-6ca1-43ce-bea8-515cdf29cd59",
+      "type": "metadata",
+      "attributes": {
+        "attributes": {},
+        "created_at": "2021-12-29T14:36:01.796775Z",
+        "description": "asd",
+        "image_url": "https://rarify-user-content-dev.s3.amazonaws.com/2b3a17df-932d-4421-b03c-b05cebc7eae5",
```
Comment (napalmpapalam) [inline packages/json-api-client/src/test/mocks/json-api-response-without-links-raw.json]: /

File: packages/json-api-client/src/types/json-api.ts [hunk 50]
```diff
@@ -0,0 +1,84 @@
...
+
+export type JsonApiResponseLinks = {
+  first?: Endpoint
+  last?: Endpoint
+  next?: Endpoint
+  prev?: Endpoint
+  self?: Endpoint
+}
+
+export type RequestOpts = {
```
Comment (napalmpapalam) [inline packages/json-api-client/src/types/json-api.ts]: not consistent type naming => `JsonApiClientRequestOpts`

**Review by napalmpapalam — CHANGES_REQUESTED** [review]

File: packages/json-api-client/src/browser.ts (hunk 29 above)
Comment (Sorizen) [reply packages/json-api-client/src/browser.ts]: 100% done)

File: packages/json-api-client/package.json (hunk 30 above)
Comment (Sorizen) [reply packages/json-api-client/package.json]: done

File: packages/json-api-client/src/json-api.ts (hunk 32 above)
Comment (Sorizen) [reply packages/json-api-client/src/json-api.ts]: done

File: packages/json-api-client/src/test/mocks/json-api-response-parsed.json (hunk 33 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-parsed.json]: done, changet to testnet

File: packages/json-api-client/src/test/mocks/json-api-response-parsed.json (hunk 34 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-parsed.json:140]: done

File: packages/json-api-client/src/test/mocks/json-api-response-parsed.json (hunk 35 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-parsed.json]: done

File: packages/json-api-client/src/test/mocks/json-api-response-parsed.json (hunk 36 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-parsed.json]: changed to test@test.com

File: packages/json-api-client/src/test/mocks/json-api-response-parsed.json (hunk 37 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-parsed.json]: done

File: packages/json-api-client/src/test/mocks/json-api-response-parsed.json (hunk 38 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-parsed.json]: done

File: packages/json-api-client/src/test/mocks/json-api-response-parsed.json (hunk 39 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-parsed.json]: done

File: packages/json-api-client/src/test/mocks/json-api-response-raw.json (hunk 40 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-raw.json]: done

File: packages/json-api-client/src/test/mocks/json-api-response-raw.json (hunk 41 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-raw.json]: done

File: packages/json-api-client/src/test/mocks/json-api-response-raw.json (hunk 42 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-raw.json]: done

File: packages/json-api-client/src/test/mocks/json-api-response-raw.json (hunk 43 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-raw.json]: changed to tetnet

File: packages/json-api-client/src/test/mocks/json-api-response-raw.json (hunk 44 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-raw.json]: done

File: packages/json-api-client/src/test/mocks/json-api-response-raw.json (hunk 45 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-raw.json]: done

File: packages/json-api-client/src/test/mocks/json-api-response-without-links-raw.json (hunk 46 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-without-links-raw.json]: done

File: packages/json-api-client/src/test/mocks/json-api-response-without-links-raw.json (hunk 47 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-without-links-raw.json]: done

File: packages/json-api-client/src/test/mocks/json-api-response-without-links-raw.json (hunk 48 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-without-links-raw.json]: done

File: packages/json-api-client/src/test/mocks/json-api-response-without-links-raw.json (hunk 49 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-without-links-raw.json]: done

File: packages/json-api-client/src/test/mocks/json-api-response-parsed.json (hunk 34 above)
Comment (Sorizen) [reply packages/json-api-client/src/test/mocks/json-api-response-parsed.json]: changed to test@test.com

File: packages/json-api-client/src/types/json-api.ts (hunk 50 above)
Comment (Sorizen) [reply packages/json-api-client/src/types/json-api.ts]: done

**Review by napalmpapalam — APPROVED** [review]

**Pull request merged**